Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/InteropCodeGenerator/SpatialGDKEditorInteropCodeGenerator.cpp

```diff
@@ -127,36 +119,110 @@ FString GenerateIntermediateDirectory()
 	return AbsoluteCombinedIntermediatePath;
 }
 
-bool SpatialGDKGenerateInteropCode()
+void CreateSchemaDatabase(TArray<UClass*> Classes)
 {
-	const USpatialGDKEditorToolbarSettings* SpatialGDKToolbarSettings = GetDefault<USpatialGDKEditorToolbarSettings>();
-	TArray<UClass*> InteropGeneratedClasses;
+	AsyncTask(ENamedThreads::GameThread, [Classes]{
+		FString PackagePath = TEXT("/Game/Spatial/SchemaDatabase");
+
+		UPackage *Package = CreatePackage(nullptr, *PackagePath);
+
+		USchemaDatabase* SchemaDatabase = NewObject<USchemaDatabase>(Package, USchemaDatabase::StaticClass(), FName("SchemaDatabase"), EObjectFlags::RF_Public | EObjectFlags::RF_Standalone);
+
+		int ComponentId = 100010;
+		for (UClass* Class : Classes)
+		{
+			FSchemaData SchemaData;
+			SchemaData.SingleClientRepData = ComponentId++;
+			SchemaData.MultiClientRepData = ComponentId++;
+			SchemaData.HandoverData = ComponentId++;
+			SchemaData.ClientRPCs = ComponentId++;
+			SchemaData.ServerRPCs = ComponentId++;
+			SchemaData.CrossServerRPCs = ComponentId++;
+			SchemaData.NetMulticastRPCs = ComponentId++;
+
+			SchemaDatabase->ClassToSchema.Add(Class, SchemaData);
+		}
+
+		FAssetRegistryModule::AssetCreated(SchemaDatabase);
+		SchemaDatabase->MarkPackageDirty();
+
+		FString FilePath = FString::Printf(TEXT("%s%s"), *PackagePath, *FPackageName::GetAssetPackageExtension());
+		bool bSuccess = UPackage::SavePackage(Package, SchemaDatabase, EObjectFlags::RF_Public | EObjectFlags::RF_Standalone, *FPackageName::LongPackageNameToFilename(PackagePath, FPackageName::GetAssetPackageExtension()));
+	});
+}
+
+TArray<UClass*> GetAllSpatialTypeClasses()
+{
+	TArray<UClass*> Classes;
+
 	for (TObjectIterator<UClass> It; It; ++It)
 	{
 		if (It->HasAnySpatialClassFlags(SPATIALCLASS_GenerateTypeBindings) == false)
 		{
 			continue;
 		}
 
-		// Ensure we don't process skeleton or reinitialised classes
-		if (	It->GetName().StartsWith(TEXT("SKEL_"), ESearchCase::CaseSensitive)
-			||	It->GetName().StartsWith(TEXT("REINST_"), ESearchCase::CaseSensitive))
+		// Ensure we don't process skeleton or reinitialized classes
+		if (It->GetName().StartsWith(TEXT("SKEL_"), ESearchCase::CaseSensitive)
+			|| It->GetName().StartsWith(TEXT("REINST_"), ESearchCase::CaseSensitive))
 		{
 			continue;
 		}
 
-		InteropGeneratedClasses.Add(*It);
+		Classes.Add(*It);
 	}
 
+	return Classes;
+}
+
+TArray<UClass*> GetAllSupportedClasses()
```

Comment:
After a scan in testsuite and the GDK, I cannot find any references to this function, where is it used?

Reply:
This is a dormant function for convenience for myself and Midwinter.